### PR TITLE
statistics: move interfaces out of the util package

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close None

Problem Summary:

### What is changed and how it works?

- Added a pool interface.

- Added a tableInfoGetter interface.

- Moved sysProcTracker to StatsAnalyze, we don't need to keep it in the handler.

- Moved all interfaces from util to types pckage.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/types/interfaces.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package util
+package types
```
Comment (hawkingrei) [inline pkg/statistics/handle/types/interfaces.go:15]:
```suggestion
package handleimpl
```

It is better to call ```handleimpl```.

**Review by hawkingrei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**